Adding topic about switching SaaS environments (#526)
magento/merchdocs: src/marketing/recommendation-change-source.md

---
title: Fetch Product Recommendations from a Different Source
tag: product-recommendations
ee_only: true
---

> A Magento store configured to use product recommendations fetches those recommendations through a [SaaS service]({% link configuration/services/saas.md %}) linked to your production environment.

Reviewer: Needs context along the lines of: 

When you configure Recommendations with a SaaS Env, that SaaS Env collects catalog data and storefront behavioral data. Adobe Sensei analyzes this data collected and computes product associations used to serve Product Recommendations. 

The challenge is that on non-prod envs like testing or staging, there is often not much storefront data (unless automated) to feed Recommendations. Real shopper behavior at scale is captured only on the production env. 

Magento allows you to use the "learnings" based on shopper events in your production environment on other, non-prod envs. This allows you to preview which recommendations your shoppers will see and experiment with different recommendation types and placement locations.

Through the Admin UI in your testing environment, you can fetch recommendations generated from actual shopper data from your production environment or another environment. Changing where those recommendations are fetched is done by specifying a different SaaS environment.

Requirements: 
1. Storefront data collection is configured and enabled on production
2. Catalog on your non-prod envs is largely the same as you have on production. Staging is a good candidate because it's a best practice to have your catalog there be analogous to your production catalog. 

Important: when recommendations from another saas env are displayed on storefront, they will not be clickable. They are served for preview purposes only.

---
title: Fetch Product Recommendations from a Different Source
tag: product-recommendations
ee_only: true
---

When you [configure a SaaS environment]({% link configuration/services/saas.md %}) for Recommendations, that SaaS environment collects catalog data and storefront behavioral data. [Adobe Sensei](https://www.adobe.com/sensei.html) analyzes that data and computes product associations used to serve Product Recommendations.

On non-production environments like testing or staging, there is often not much storefront behavioral data (unless automated) to serve Recommendations. Actual shopper behavior at scale is captured only on the production environment.

> Magento allows you to use the storefront behavioral data from your production environment on other, non-production environments. Using this data inside non-production environments allows you to preview which recommendations your shoppers will see and experiment with different recommendation types and placement locations.

Reviewer: "Magento allows you to use product recommendations computed in your product environment on other, non-prod.... "
(we don't use behavioral data directly)

Through the Admin UI in your non-production environment, you can fetch recommendations generated from actual shopper data from your production environment or another environment. Changing where those recommendations are fetched is done by specifying a different SaaS environment.

{:.bs-callout-info}

> When recommendations from another SaaS environment are displayed on the storefront, you can only preview them, they are not clickable.

Reviewer 2: ```suggestion
When recommendations from another SaaS environment are displayed on the storefront, a storefront user can only preview them (they are not clickable).
```

> ## Fetch Recommendations from Different SaaS Environment

Reviewer 2: ```suggestion
## Fetch Recommendations from different SaaS environment
```

Before you begin, make sure that:

> -  Storefront data collection is configured and enabled on production.

Reviewer: add a link to verifying storefront data collection

> -  Your non-production environment catalog is largely the same as you have on production. Having similar catalogs ensures that the products returned in the recommendation units closely mimic those on production.

Reviewer 2: ```suggestion
-  Your non-production environment catalog is largely the same as the one that you have in production. Using similar catalogs ensures that the products returned in the recommendation units closely mimic those on production.
```

1. Launch your non-production environment.

> 1. On the _Admin_ sidebar, go to **Marketing** > _Promotions_ > **Product Recommendations** to display the **Product Recommendations** dashboard.

Reviewer 2: ```suggestion
1. On the _Admin_ sidebar, go to **Marketing** > _Promotions_ > **Product Recommendations**.
```

1. Select **Settings**.

> 1. In the **Recommendations source** section, enable the **Fetch recommendations from a different SaaS environment** switch.

Reviewer 2: We have been using italic for items on the screen that the user does not "touch". This helps to reduce the use or bold. 

```suggestion
1. In the _Recommendations source_ section, enable the **Fetch recommendations from a different SaaS environment** option.
```

>    The **Available SaaS Environments** drop-down list appears.

Reviewer 2: ```suggestion
   The **Available SaaS Environments** list appears.
```

1. Select the SaaS environment associated with your production environment or another environment that has shopper data you want to use.

1. Click **Save changes**.

>    Magento will now fetch recommendations from that environment.

Reviewer: - need to add a note that recommended items on storefront won't be clickable
